R/read_vpts.R

```diff
@@ -1,241 +1,144 @@
-  }
-  # currently only two delimitors supported
-  # "" for legacy vol2bird output, "," for csv output
-  assertthat::assert_that(assertthat::is.scalar(sep) && is.character(sep),
-                          msg = "'sep' should be either \",\" or \"\"")
-  assertthat::assert_that(sep == "" || sep == ",",
-                          msg = "'sep' should be either \",\" or \"\"")
-
-  if (!missing(lat)) {
-    assertthat::assert_that(
-      assertthat::is.number(lat),
-      msg = "'lat' should be a single numeric between -90 and 90 degrees"
+read_vpts <- function(files, ...) {
+
+  #Define valid extensions
+  valid_extensions <- c("csv", "gz", "h5", "txt")
+
+  # Get file extension
+  extension <- unique(tools::file_ext(files))
+
+  assertthat::assert_that(
+    length(extension) == 1,
+    msg = "`files` must all have the same extension."
+  )
+
+  # If the file has an extension, check if it is valid
+  if (extension != "") {
+      assertthat::assert_that(
+      extension %in% valid_extensions,
+      msg = glue::glue(
+        "`files` must have one of the following extensions: {valid_extensions_collapse}",
+        valid_extensions_collapse = glue::glue_collapse(valid_extensions, sep = ", ")
       )
-    if (lat < -90 || lat > 90) {
-      stop("'lat' should be numeric between -90 and 90 degrees")
-    }
-  }
-  if (!missing(lon)) {
-    assertthat::assert_that(
-      assertthat::is.number(lon),
-      msg = "'lon' should be a single numeric numeric between -360 and 360 degrees"
     )
-    if (lon < -360 || lon > 360) {
-      stop("'lon' should be numeric between -360 and 360 degrees")
-    }
-  }
-  if (!missing(height)) {
-    assertthat::assert_that(assertthat::is.number(height))
-    if (!is.numeric(height) || height < 0) {
-      stop("'height' should be a positive number of meters above sea level")
-    }
-  }
-  if (missing(radar) && sep =="") {
-    stop("'radar' argument missing. Required to specify a radar identifier.")
-  }
-  if (missing(wavelength)) {
-    warning(paste("No 'wavelength' argument provided, assuming radar operates",
-      " at ", wavelength, "-band",
-      sep = ""
-    ))
+  } else {
+    # If the file does not have an extension, infer the file type
+    extension <- guess_file_type(files)
   }
-  wavelength_msg <-
-    glue::glue(
-      "'wavelength' should be a single positive number",
-      ", or one of 'C' or 'S' for C-band and S-band radar, respectively."
-    )
-  assertthat::assert_that(
-    (assertthat::is.number(wavelength) && wavelength > 0) ||
-      (assertthat::is.scalar(wavelength) && wavelength %in% c("C","S")),
-    msg = wavelength_msg)
 
-  if (wavelength == "C") {
-    wavelength <- 5.3
-  }
-  if (wavelength == "S") {
-    wavelength <- 10.6
+  # Check if the input file has a .txt extension and if so reroute to read_stdout
+  if (extension == "txt") {
+    warning(".txt extenstion detected - falling back to read_stdout(). The use of read_stdout() will be deprecated soon.
```

Comment:
I don't think we need to deprecate read_stdout, we can keep the functionality for quite a while. 

Suggested behaviour for .txt extension:
1. verify if comma delimited or not
2. if comma delimited, try reading VPTS CSV
3. if not comma delimited, try reading stdout format

So allow user to name their csv file with a txt extension or without any extension